pkg/leader/leader.go

```diff
@@ -43,42 +51,75 @@ func run(ctx context.Context, namespace, name string, client kubernetes.Interfac
 		logrus.Fatalf("error creating leader lock for %s: %v", name, err)
 	}
 
-	t := time.Second
-	if dl := os.Getenv("CATTLE_DEV_MODE"); dl != "" {
-		t = time.Hour
+	cbs := leaderelection.LeaderCallbacks{
+		OnStartedLeading: func(ctx context.Context) {
+			go cb(ctx)
+		},
+		OnStoppedLeading: func() {
+			select {
+			case <-ctx.Done():
+				// The context has been canceled or is otherwise complete.
+				// This is a request to terminate. Exit 0.
+				// Exiting cleanly is useful when the context is canceled
+				// so that Kubernetes doesn't record it exiting in error
+				// when the exit was requested. For example, the wrangler-cli
+				// package sets up a context that cancels when SIGTERM is
+				// sent in. If a node is shut down this is the type of signal
+				// sent. In that case you want the 0 exit code to mark it as
+				// complete so that everything comes back up correctly after
+				// a restart.
+				// The pattern found here can be found inside the kube-scheduler.
+				logrus.Info("requested to terminate, exiting")
+				os.Exit(0)
+			default:
+				logrus.Fatalf("leaderelection lost for %s", name)
+			}
+		},
 	}
 
-	leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
-		Lock:          rl,
-		LeaseDuration: 45 * t,
-		RenewDeadline: 30 * t,
-		RetryPeriod:   2 * time.Second,
-		Callbacks: leaderelection.LeaderCallbacks{
-			OnStartedLeading: func(ctx context.Context) {
-				go cb(ctx)
-			},
-			OnStoppedLeading: func() {
-				select {
-				case <-ctx.Done():
-					// The context has been canceled or is otherwise complete.
-					// This is a request to terminate. Exit 0.
-					// Exiting cleanly is useful when the context is canceled
-					// so that Kubernetes doesn't record it exiting in error
-					// when the exit was requested. For example, the wrangler-cli
-					// package sets up a context that cancels when SIGTERM is
-					// sent in. If a node is shut down this is the type of signal
-					// sent. In that case you want the 0 exit code to mark it as
-					// complete so that everything comes back up correctly after
-					// a restart.
-					// The pattern found here can be found inside the kube-scheduler.
-					logrus.Info("requested to terminate, exiting")
-					os.Exit(0)
-				default:
-					logrus.Fatalf("leaderelection lost for %s", name)
-				}
-			},
-		},
-		ReleaseOnCancel: true,
-	})
+	config, err := computeConfig(rl, cbs)
+	if err != nil {
+		return err
+	}
+
+	leaderelection.RunOrDie(ctx, *config)
 	panic("unreachable")
 }
+
+func computeConfig(rl resourcelock.Interface, cbs leaderelection.LeaderCallbacks) (*leaderelection.LeaderElectionConfig, error) {
+	leaseDuration := defaultLeaseDuration
+	renewDeadline := defaultRenewDeadline
+	retryPeriod := defaultRetryPeriod
+	var err error
+	if d := os.Getenv("CATTLE_DEV_MODE"); d != "" {
```

Comment:
Nit: I know you didn't add this, but while we're here, can you move these string values to package level const.
```suggestion
	if d := os.Getenv(devModeKey); d != "" {
```